Drop bearer_token before authenticating

uncleNight opened the pull request:
Long story short - if a token has expired, obtaining a new one is impossible because `client.engine.http()` prioritizes `bearer_token` over basic HTTP auth - but all the `/v2/account/authenticate/*` endpoints need HTTP auth and not a token.

Decided to not follow the "check if expired" route and simply drop `bearer_token` from `client.config` in every `nakama.authenticate_*` method.

Two kinda related questions left:
* `nakama.nakama` module begins with `-- Code generated by codegen/main.go. DO NOT EDIT.` - I believe that was a one-time thing and this comment can be removed now? Don't really think further code generation is going to be used for improving this module;
* every `nakama.authenticate_*` method checks for `api_session` in a callback/coroutine, but the variable is never declared and basically always `nil`. Set `local api_session = require "nakama.session"` because it seems like what's missing here, let me know please if my assumption is wrong.

britzl commented:
> Decided to not follow the "check if expired" route and simply drop bearer_token from client.config in every nakama.authenticate_* method.

Yep, good one!

> nakama.nakama module begins with -- Code generated by codegen/main.go. DO NOT EDIT. - I believe that was a one-time thing and this comment can be removed now? Don't really think further code generation is going to be used for improving this module

Well, actually now that I think of it, any change done to nakama.lua should be done to the template in codegen/ and not directly in nakama.lua. This means that this fix for http/https selection is incorrect:

https://github.com/heroiclabs/nakama-defold/commit/af61839507de6cfaddc84ccf7369a5ed15b25e2d

We don't want to make changes to nakama.lua since that will make future updates a lot harder if/when HeroicLabs updates the API spec.

> every nakama.authenticate_* method checks for api_session in a callback/coroutine, but the variable is never declared and basically always nil. Set local api_session = require "nakama.session" because it seems like what's missing here, let me know please if my assumption is wrong.

Indeed, you are absolutely correct! "nakama.session" should be required and used!

uncleNight commented:
Thanks for explaining, I got so focused on the Lua part that I forgot that the codegen script was there in the first place.

Updated the template (added an extra template function since `text/template` doesn't support regex or startWith statements, or I'm simply not aware of that), regenerated `nakama.lua` based on the new template and up-to-date Nakama code. Fixed http/https selection while I'm at it.

britzl commented:
Great job! Thanks for the contribution!
